commercetools-models/src/test/java/io/sphere/sdk/messages/queries/MessageQueryIntegrationTest.java

```diff
@@ -161,11 +169,16 @@ public void queryForASpecificMessage() throws Exception {
                             .withExpansionPaths(m -> m.resource())
                             .withLimit(1L)
                             .forMessageType(ReturnInfoAddedMessage.MESSAGE_HINT);
-            final PagedQueryResult<ReturnInfoAddedMessage> pagedQueryResult = client().executeBlocking(query);
-            final ReturnInfoAddedMessage message = pagedQueryResult.head().get();
-            assertThat(message.getReturnInfo()).isEqualTo(returnInfo);
-            assertThat(message.getResource().getObj()).isNotNull();
-            assertThat(message.getResource().getId()).isEqualTo(order.getId());
+
+            assertEventually(() -> {
+                final PagedQueryResult<ReturnInfoAddedMessage> pagedQueryResult = client().executeBlocking(query);
+                final Optional<ReturnInfoAddedMessage> message = pagedQueryResult.head();
+
+                assertThat(message).isPresent();
+                assertThat(message.get().getReturnInfo()).isEqualTo(returnInfo);
```

Comment:
the result of `message.get()` could have been saved into a variable here, but it is good enough to merge.